### PR TITLE
Fix incorrect and inconsistent timeout types

### PR DESCRIPTION
**Pull request by revolter**

### CODE REVIEW

**Comment by Bibo-Joshi** [comment]
Hi. I'm not quite sure what this PR is aiming at. Would you mind explaining where you see inconsistency in the types? IISC we use `float` everywhere.

**Comment by revolter** [comment]
Yes, sure, sorry for not explaining it from the start.

So, knowing that the [official `getUpdates` API](https://core.telegram.org/bots/api#getupdates) declares `timeout` as `Integer`, here are some of the inconsistencies:
1. The `Updater.start_polling` correctly declares it as `int`:

   https://github.com/python-telegram-bot/python-telegram-bot/blob/57c780c62f5c1a3ee8441d6b690dd3f5a3d22288/telegram/ext/_updater.py#L187

   but it's incorrectly documented as `float`:

   https://github.com/python-telegram-bot/python-telegram-bot/blob/57c780c62f5c1a3ee8441d6b690dd3f5a3d22288/telegram/ext/_updater.py#L205
2. The `Bot.get_updates` incorrectly declares it as `float`:

   https://github.com/python-telegram-bot/python-telegram-bot/blob/57c780c62f5c1a3ee8441d6b690dd3f5a3d22288/telegram/_bot.py#L3589

   although it's correctly documented as `int`:

   https://github.com/python-telegram-bot/python-telegram-bot/blob/57c780c62f5c1a3ee8441d6b690dd3f5a3d22288/telegram/_bot.py#L3624

So, the bottom line is that all of them should be `int`, which is what this PR does.

File: telegram/_bot.py [hunk 1]
```diff
@@ -3661,7 +3661,7 @@ async def get_updates(
             await self._post(
                 "getUpdates",
                 data,
-                read_timeout=read_timeout + timeout if timeout else read_timeout,
+                read_timeout=read_timeout + float(timeout) if timeout else read_timeout,
```
Comment (Bibo-Joshi) [inline telegram/_bot.py:3664]: float + integer should work without explicit conversion, shoudn't it? 🤔

**Review by Bibo-Joshi — COMMENTED** [review]
Thanks for the explanation and the catch! Yes, I would like to have the merged, then :) Would you mind resolving the merge conflicts?

File: telegram/_bot.py (hunk 1 above)
Comment (revolter) [reply telegram/_bot.py:3664]: I don't know, actually 😅 I'm coming from Swift, and I thought it doesn't work like that. So I'll revert this change 👍🏻

**Comment by Bibo-Joshi** [comment]
Than you for the contribution!

**Comment by revolter** [comment]
And thank you for merging!

**Pull request merged**